### PR TITLE
fix(Info): Fix token image

### PR DESCRIPTION
**Pull request by Woody810**
---

## PR-Codex overview
This PR updates the `CurrencyLogo` component to dynamically fetch token logos using new methods.

### Detailed summary
- Imported `useHttpLocations`, `Currency`, and `Token`
- Removed `multiChainId` import
- Updated logo retrieval logic
- Utilized `useMemo` for logo sources
- Updated `StyledLogo` component props
- Removed unused imports and variables

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

Before:
image

After:
image

### CODE REVIEW

File: apps/web/src/views/Info/components/CurrencyLogo/index.tsx [hunk 1]
```diff
@@ -28,7 +28,22 @@ export const CurrencyLogo: React.FC<
...
+    case 'arb':
+      imagePath = 'arbitrum/'
+      break
+    case 'polygon_zkevm':
+      imagePath = 'polygon-zkevm/'
+      break
+    default:
+      imagePath += '/'
+      break
+  }
```
Comment (ChefMomota) [inline apps/web/src/views/Info/components/CurrencyLogo/index.tsx:45]: Avoid duplicate code.
Try use `getImageUrlFromToken`

Comment (Woody810) [reply apps/web/src/views/Info/components/CurrencyLogo/index.tsx]: fixed

Comment (ChefMomota) [reply apps/web/src/views/Info/components/CurrencyLogo/index.tsx]: You should not remove the image for `trustwallet`.
We need both, trustwallet image and ourself image url.

File: apps/web/src/views/Info/components/CurrencyLogo/index.tsx [hunk 2]
```diff
@@ -24,16 +24,11 @@ export const CurrencyLogo: React.FC<
     chainName?: MultiChainName
   }>
 > = ({ address, size = '24px', chainName = 'BSC', ...rest }) => {
-  const src = useMemo(() => {
-    return getTokenLogoURL(new Token(multiChainId[chainName], address as Address, 18, ''))
-  }, [address, chainName])
```
Comment (ChefMomota) [inline apps/web/src/views/Info/components/CurrencyLogo/index.tsx:29]: You should not remove the image for trustwallet.
We need both, trustwallet image and ourself image url.

File: apps/web/src/views/Info/components/CurrencyLogo/index.tsx [hunk 3]
```diff
@@ -24,15 +25,15 @@ export const CurrencyLogo: React.FC<
...
 
-  const imagePath = chainName === 'BSC' ? '' : `${chainName?.toLowerCase()}/`
   const checkedsummedAddress = safeGetAddress(address)
-  const srcFromPCS = checkedsummedAddress
-    ? `https://tokens.pancakeswap.finance/images/${imagePath}${checkedsummedAddress}.png`
-    : ''
+  const srcFromPCS = useMemo(() => {
+    return getImageUrlFromToken(new Token(multiChainId[chainName], checkedsummedAddress as Address, 18, ''))
+  }, [checkedsummedAddress, chainName])
   return <StyledLogo size={size} srcs={[srcFromPCS, src]} alt="token logo" useFilledIcon {...rest} />
```
Comment (chef-jojo) [inline apps/web/src/views/Info/components/CurrencyLogo/index.tsx:37]: Usememo for srcs here
You can combine those into one

**Review by chef-jojo — CHANGES_REQUESTED** [review]

File: apps/web/src/views/Info/components/CurrencyLogo/index.tsx [hunk 4]
```diff
@@ -24,16 +25,15 @@ export const CurrencyLogo: React.FC<
...
-    return getTokenLogoURL(new Token(multiChainId[chainName], address as Address, 18, ''))
-  }, [address, chainName])
-
-  const imagePath = chainName === 'BSC' ? '' : `${chainName?.toLowerCase()}/`
   const checkedsummedAddress = safeGetAddress(address)
-  const srcFromPCS = checkedsummedAddress
-    ? `https://tokens.pancakeswap.finance/images/${imagePath}${checkedsummedAddress}.png`
-    : ''
-  return <StyledLogo size={size} srcs={[srcFromPCS, src]} alt="token logo" useFilledIcon {...rest} />
+  const srcs = useMemo(() => {
```
Comment (chefjackson) [inline apps/web/src/views/Info/components/CurrencyLogo/index.tsx:29]: Can we use `getCurrencyLogoUrls` from `widgets-internal`?

**Comment by ChefMomota** [comment]
Pls check it again
Screenshot 2024-03-05 at 2 56 05 PM

File: apps/web/src/views/Info/components/CurrencyLogo/index.tsx [hunk 5]
```diff
@@ -20,20 +19,30 @@ export const CurrencyLogo: React.FC<
   React.PropsWithChildren<{
     address?: string
     token?: Token
+    currency?: Currency & {
```
Comment (chefjackson) [inline apps/web/src/views/Info/components/CurrencyLogo/index.tsx:22]: I don't think the currency is defined here? For info page we may not be able to get the currency/token instance. We probably need to add more utility functions like `getTokenLogoUrlsByAddress` to the widget. Like what we did here https://github.com/pancakeswap/pancake-frontend/blob/develop/packages/widgets-internal/components/CurrencyLogo/utils.ts#L30-L40

**Comment by ChefJerry** [comment]
Close this PR because it has been inactive for a long time. Please reopen if you still intend to submit this pull request.

**Pull request closed without merging**